Solutions/Marain.Tenancy.Cli/Marain/Tenancy/Cli/Commands/List.cs

```diff
@@ -0,0 +1,147 @@
+            ShortName = "t",
+            LongName = "tenant",
+            Description = "The Id of the tenant to retrieve children for. Leave blank to retrieve children of the root tenant.")]
+        public string? TenantId { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value containing the names of specific properties that should be included in the output.
+        /// </summary>
+        [Option(
+            CommandOptionType.MultipleValue,
+            ShortName = "p",
+            LongName = "property",
+            Description = "The names of tenant properties to include in the output. If omitted, only the tenant Ids will be listed.")]
+#pragma warning disable SA1011 // Closing square brackets should be spaced correctly
+        public string[]? IncludeProperties { get; set; }
+#pragma warning restore SA1011 // Closing square brackets should be spaced correctly
+
+        /// <summary>
+        /// Executes the command.
+        /// </summary>
+        /// <param name="app">The current <c>CommandLineApplication</c>.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task OnExecute(CommandLineApplication app)
+        {
+            if (string.IsNullOrEmpty(this.TenantId))
+            {
+                this.TenantId = this.tenantProvider.Root.Id;
+            }
+
+            string? continuationToken = null;
+
+            var childTenantIds = new List<string>();
+
+            do
+            {
+                TenantCollectionResult children = await this.tenantProvider.GetChildrenAsync(
+                    this.TenantId,
+                    20,
+                    continuationToken).ConfigureAwait(false);
+
+                childTenantIds.AddRange(children.Tenants);
+
+                continuationToken = children.ContinuationToken;
+            }
+            while (!string.IsNullOrEmpty(continuationToken));
+
+            if (this.IncludeProperties?.Length > 0)
+            {
+                await this.LoadAndOutputTenantDetailsAsync(childTenantIds, app.Out).ConfigureAwait(false);
+            }
+            else
+            {
+                this.OutputTenantIds(childTenantIds, app.Out);
+            }
+        }
+
+        private async Task LoadAndOutputTenantDetailsAsync(List<string> children, TextWriter output)
+        {
+            IEnumerable<Task<ITenant>> detailsTasks = children.Select(x => this.tenantProvider.GetTenantAsync(x));
+
+            await Task.WhenAll(detailsTasks).ConfigureAwait(false);
+
+            var headings = new List<string> { "Id" };
+
+            if (this.IncludeProperties != null)
+            {
+                headings.AddRange(this.IncludeProperties);
+            }
+
+            var table = new ConsoleTable(headings.ToArray());
+
+            table.Options.OutputTo = output;
+            table.Options.EnableCount = false;
+
+            detailsTasks.ForEach(task =>
```

Comment:
...so if you do my suggestion above this would become something like `tenants.ForEach(tenant =>` and then you could lose the first line of the body.

But I have another question: why the `ForEach` extension method here? Any particular reason not to use `foreach`?